android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -208,6 +321,43 @@ public void onPageScrollStateChanged(int state) {}
         });
         showWidgetBasedOnOrder();
         NTPUtil.showBREBottomBanner(this);
+
+        if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_NEWS)) {
+            SharedPreferences sharedPreferences = ContextUtils.getAppSharedPreferences();
+
+            isNewsOn = BravePrefServiceBridge.getInstance().getNewsOptIn();
+            isShowOptin = sharedPreferences.getBoolean(BraveNewsPreferences.PREF_SHOW_OPTIN, false);
+            isShowNewsOn = BravePrefServiceBridge.getInstance().getShowNews();
+
+            feedHash = "";
+            isFeedUpdateAvailable = false;
+
+            InitBraveNewsController();
+            braveNewsUtils = new BraveNewsUtils();
+
+            isFeedLoaded = BraveActivity.getBraveActivity().isLoadedFeed();
+            existingNewsFeedObject = BraveActivity.getBraveActivity().getNewsItemsFeedCards();
+            prevScrollPosition = 2;
+
+            if (isNewsOn && isShowNewsOn) {
```

Comment:
maybe combine into 
`if (isNewsOn && isShowNewsOn && isFeedLoaded && existingNewsFeedObject != null) {`
?

Reply:
thanks. it's replaced

Reply:
:+1: